app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  include SessionsHelper
+
+  def failure
+    flash[:alert] = 'ログインをキャンセルしました。もう一度ログインしてください'
+    redirect_to root_path
+  end
+  
+  def create
+    user_data = request.env['omniauth.auth']
+    binding.pry
+    user = User.find_or_initialize_by(uid: user_data[:uid])
+    user.update_attributes(nickname: user_data[:info][:nickname])
```

Comment:
1.  `update_attributes`は将来廃止対象なので、`update`を使用しましょう。
 ※実行ログを注意深く見ると以下のwarningがあります。

```
DEPRECATION WARNING: update_attributes is deprecated and will be removed from Rails 6.1 (please, use update instead)
```

2. userを「find_or_initializeした後、値を更新する」という２つの処理を連続して実施する場合には、メソッドチェーンを使うとコードがスッキリします。

```suggestion
    user = User.find_or_initialize_by(XXXX).update(XXXX)
```